network/p2p/ping.go

```diff
@@ -0,0 +1,174 @@
+package p2p
+
+import (
+	"context"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	"github.com/libp2p/go-msgio"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/cmd/bootstrap/build"
+	"github.com/onflow/flow-go/network/message"
+)
+
+// the Flow Ping protocol prefix
+const FlowLibP2PPingPrefix = "/flow/ping/"
+
+const pingTimeout = time.Second * 60
+
+// PingService handles the outbound and inbound ping requests and response
+type PingService struct {
+	host           host.Host
+	logger         zerolog.Logger
+	pingProtocolID protocol.ID
+}
+
+func NewPingService(h host.Host, pingProtocolID protocol.ID, logger zerolog.Logger) *PingService {
+	ps := &PingService{host: h, logger: logger, pingProtocolID: pingProtocolID}
+	h.SetStreamHandler(pingProtocolID, ps.PingHandler)
+	return ps
+}
+
+// PingHandler receives the inbound stream for Flow ping protocol and respond back with the PingResponse message
+func (ps *PingService) PingHandler(s network.Stream) {
+
+	errCh := make(chan error, 1)
+	defer close(errCh)
+	timer := time.NewTimer(pingTimeout)
+	defer timer.Stop()
+
+	log := ps.logger.With().Str("remote", s.Conn().RemoteMultiaddr().String()).Str("protocol", string(s.Protocol())).Logger()
+
+	go func() {
+		select {
+		case <-timer.C:
+			// if read or write took longer than configured timeout, then reset the stream
+			log.Error().Msg("ping timeout")
+			err := s.Reset()
+			if err != nil {
+				log.Error().Err(err).Msg("failed to reset stream")
+			}
+		case err, ok := <-errCh:
+			// if and error occur while reposning to a pind request, then reset the stream
+			if ok {
+				log.Error().Err(err)
```

Comment:
```suggestion
				log.Error().Err(err).Msg("error responding to ping request")
```
I don't think this line will print anything as-is, believe it needs the `Msg` call as well